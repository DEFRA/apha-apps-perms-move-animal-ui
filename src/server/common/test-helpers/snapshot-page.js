import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
