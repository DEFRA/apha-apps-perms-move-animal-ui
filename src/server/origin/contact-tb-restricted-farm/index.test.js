import { describePageSnapshot } from '../../common/test-helpers/snapshot-page.js'
import { OriginContactTbRestrictedFarmPage } from './index.js'

const pageUrl = '/origin/contact-the-tb-restricted-farm'

describe('ContactTbRestrictedFarmPage', () => {
  it('should have the correct properties', () => {
    const page = new OriginContactTbRestrictedFarmPage()
    expect(page.sectionKey).toBe('origin')
    expect(page.pageTitle).toBe(
      'You need to contact the TB restricted farm the animals are moving onto'
    )
    expect(page.urlPath).toBe(pageUrl)
    expect(page.view).toBe('origin/contact-tb-restricted-farm/index')
  })

  describePageSnapshot({
    describes: '#ContactTbRestrictedFarmPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
