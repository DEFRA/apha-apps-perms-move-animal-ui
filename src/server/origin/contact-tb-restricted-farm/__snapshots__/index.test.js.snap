// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTbRestrictedFarmPage #ContactTbRestrictedFarmPage.content should render expected response and content 1`] = `
"
  <div class="govuk-grid-row app-heading" data-testid="app-heading">
  <div class="govuk-grid-column-two-thirds">
    <h1 class="govuk-heading-xl govuk-!-margin-bottom-2" data-testid="app-heading-title">
      500
    </h1>

  </div>
</div>


  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <p>Something went wrong</p>
    </div>
  </div>
        "
`;
